kademlia/network.py

```diff
@@ -220,7 +220,8 @@ async def load_state(cls, fname, port, interface='0.0.0.0'):
         log.info("Loading state from %s", fname)
         with open(fname, 'rb') as file:
             data = pickle.load(file)
-        svr = Server(data['ksize'], data['alpha'], data['id'])
+        svr = eval(cls.__name__+"(data['ksize'], data['alpha'], data['id'])")
```

Comment:
Would this not work?
```suggestion
        svr = cls(data['ksize'], data['alpha'], data['id'])
```

Reply:
Sorry, my English is poor, so my responses and these are all machine translation
When you inherit this class, because the referenced constructor is still the inherited class
So if you modify the constructor of the inherited class, the modified constructor will not work

Reply:
```suggestion
        svr = eval(cls.__name__+"(data['ksize'], data['alpha'], data['id'])")
```

Reply:
> Would this not work?
```suggestion
        svr = eval(cls.__class__.__name__+"(data['ksize'], data['alpha'], data['id'])")
```

Reply:
In the first version, I overlooked that there may not be __name__ properties, modified here

Reply:
```python
class test1:
    def __init__(self):
          print("test1")
    def save(self):
          test1()
class test2:
    def __init__(self):
          print("test2")
    def save(self):
          eval(self.__class__.__name__+"()")
class test3(test1):
    def __init__(self):
          print("test3")
class test4(test2):
    def __init__(self):
          print("test4")
test1()
a=test2()
a.save()
a=test3()
a.save()
a=test4()
a.save()
```

Reply:
> Would this not work?

Hope you can figure it out with this example

Reply:
> Would this not work?

And your changes are incomplete, because there is a possibility that the error will be reported because there is no __call__

Reply:
I don't think your example matches this scenario - the `load_state` method is a `@classmethod`.  Here's an example that I fits a bit better, and demonstrates that even a class that extends the base `Server` can be successfully instantiated in the `load_state`:

```python
class Server:
    def __init__(self, details):
        self.details = details

    @classmethod
    def load_state(cls, details):
        svr = cls(details)
        return svr

class ExtendingServer(Server):
    def test(self):
        print('hello %s' % self.details)

s = ExtendingServer.load_state('there')
s.test()
```

When this code was written, it wasn't assumed that anyone would be extending `Server` - but I don't mind supporting that potential case.